Server/src/transport/unity_instance_middleware.py

```
@@ -104,6 +104,124 @@ def clear_active_instance(self, ctx) -> None:
+        Resolve a unity_instance string to a validated instance identifier.
+
+        Accepts:
+          - Bare port number like "6401" (stdio only) -> resolved Name@hash
+          - "Name@hash" exact match
+          - Hash prefix (unique prefix match against running instances)
+
+        Raises ValueError with a user-friendly message on failure.
+        """
+        value = value.strip()
+        if not value:
+            raise ValueError("unity_instance value must not be empty.")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)